src/PaymentGateways/Gateways/Stripe/CreditCardGateway.php

```diff
@@ -0,0 +1,144 @@
+        give_insert_payment_note($paymentData->donationId, 'Stripe Customer ID: ' . $giveStripeCustomer->get_id());
+
+        give_update_meta($paymentData->donationId, '_give_stripe_source_id', $paymentMethodId);
+        give_update_meta($paymentData->donationId, '_give_stripe_customer_id', $giveStripeCustomer->get_id());
+        give_update_meta(
+            $paymentData->donationId,
+            '_give_stripe_donation_summary',
+            give_payment_gateway_donation_summary($donationData->toArray(), false)
+        );
+
+        /**
+         * This filter hook is used to update the payment intent arguments.
+         *
+         * @since 2.5.0
+         */
+        $intent_args = apply_filters(
+            'give_stripe_create_intent_args',
+            [
+                'amount' => Money::of($paymentData->price, $paymentData->currency)->getMinorAmount(),
+                'currency' => $paymentData->currency,
+                'payment_method_types' => ['card'],
+                'statement_descriptor' => give_stripe_get_statement_descriptor(),
+                'description' => '', //give_payment_gateway_donation_summary( $donationData->toArray() ),
+                'metadata' => give_stripe_prepare_metadata($paymentData->donationId, $donationData->toArray()),
+                'customer' => $giveStripeCustomer->get_id(),
+                'payment_method' => $paymentMethodId,
+                'confirm' => true,
+                'return_url' => $paymentData->redirectUrl,
+            ]
+        );
+
+        // Send Stripe Receipt emails when enabled.
+        if ( give_is_setting_enabled( give_get_option( 'stripe_receipt_emails' ) ) ) {
+            $intent_args['receipt_email'] = $paymentData->donorInfo->email;
+        }
+
+        $intent = give( LegacyStripePaymentIntent::class )->create( $intent_args );
+
+        give_insert_payment_note( $paymentData->donationId, 'Stripe Charge/Payment Intent ID: ' . $intent->id );
+        give_insert_payment_note( $paymentData->donationId, 'Stripe Payment Intent Client Secret: ' . $intent->client_secret );
+        give_update_meta( $paymentData->donationId, '_give_stripe_payment_intent_client_secret', $intent->client_secret );
+
+        switch( $intent->status() )  {
+            case 'requires_action':
+                give_insert_payment_note( $paymentData->donationId, 'Stripe requires additional action to be fulfilled.' );
+                give_update_meta( $paymentData->donationId, '_give_stripe_payment_intent_require_action_url', $intent->nextActionRedirectUrl() );
+                return new RedirectOffsite( $intent->nextActionRedirectUrl() );
+            case 'succeeded':
+                return new PaymentProcessing( $intent->id() );
+            default:
+                throw new PaymentIntentException( sprintf( __( 'Unhandled payment intent status: %s', 'give' ), $intent->status() ) );
+        }
```

Comment:
@kjohnson it looks like you got all the logic moved over, awesome!  is there anyway we can make this monster more readable?  It's just a little hard to follow for a simpleton like myself.

Reply:
@jonwaldstein this is the more readable version 😅. I do agree that more can be done, I just found myself reverting changes a lot, so I haven't settled on a direction yet. My goal in this PR is to get the logic into the Gateway API and then refactor from there since we will have some tests in place.

Reply:
I'm probably going to introduce some `Action` classes to group blocks of code that are related, which would make this class more `Controller`-like.

Reply:
@kjohnson sounds good, actions or local methods to group blocks of code would be great - thanks!

Reply:
@jonwaldstein big refactor that moves the procedural processing to `Action` classes.